app/src/main/java/io/intrepid/contest/screens/sendinvitations/SendInvitationsActivity.java

```diff
@@ -0,0 +1,149 @@
+package io.intrepid.contest.screens.sendinvitations;
+
+import android.Manifest;
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.os.Build;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import butterknife.BindView;
+import butterknife.OnClick;
+import io.intrepid.contest.R;
+import io.intrepid.contest.base.BaseMvpActivity;
+import io.intrepid.contest.base.PresenterConfiguration;
+import io.intrepid.contest.screens.sendinvitations.selectcontacts.SelectContactsActivity;
+
+import static android.view.View.GONE;
+import static android.view.View.VISIBLE;
+
+public class SendInvitationsActivity extends BaseMvpActivity<SendInvitationsContract.Presenter>
+        implements SendInvitationsContract.View {
+
+    private static final int PERMISSIONS_REQUEST_READ_CONTACTS = 100;
+
+    @BindView(R.id.select_contacts_fab_button)
+    ImageView selectContactsButton;
+    @BindView(R.id.send_invitations_icon)
+    ImageView sendInvitationsIcon;
+    @BindView(R.id.send_invitations_intro_text_view)
+    TextView sendInvitationsIntroTextView;
+
+    private Menu menu;
+
+    public static Intent makeIntent(Context context) {
+        return new Intent(context, SendInvitationsActivity.class);
+    }
+
+    @NonNull
+    @Override
+    public SendInvitationsContract.Presenter createPresenter(PresenterConfiguration configuration) {
+        return new SendInvitationsPresenter(this, configuration);
+    }
+
+    @Override
+    protected void onViewCreated(Bundle savedInstanceState) {
+        super.onViewCreated(savedInstanceState);
+
+        setActionBarTitle(R.string.invite_contestants_bar_title);
+        setActionBarDisplayHomeAsUpEnabled(true);
+    }
+
+    @Override
+    protected int getLayoutResourceId() {
+        return R.layout.activity_send_invitations;
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        MenuInflater inflater = getMenuInflater();
+        inflater.inflate(R.menu.menu_send_invitations, menu);
+        this.menu = menu;
+        presenter.onCreateOptionsMenu();
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        presenter.onOptionsItemSelected(item.getItemId());
+        return true;
+    }
+
+    @SuppressLint("NewApi")
```

Comment:
Actually I just supress this warning because Lint doesn't realize that the second parameter will never be executed if the api < `Build.VERSION_CODES.M`
I can clarify that doing:
```
if (return Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
   return true;
}
checkSelfPermission(Manifest.permission.READ_CONTACTS) == PackageManager.PERMISSION_GRANTED);
```
But then I have to supress the optimization warning
(which might still be better for clarity)